File: functions/libsodium/variants/c/2_b64_char_to_byte_0.c
#include <stdint.h>

static unsigned int b64_char_to_byte(int c)
{
    if ('A' <= c && c <= 'Z') return c - 'A';
    if ('a' <= c && c <= 'z') return c - 'a' + 26;
    if ('0' <= c && c <= '9') return c - '0' + 52;
    if (c == '+') return 62;
    if (c == '/') return 63;
    return 0;
}


int main(){
    b64_char_to_byte(0);
}

File: functions/libsodium/variants/go/2_b64_char_to_byte_0.go
package main

func b64CharToByte(c int32) uint32 {
	const x = ((((((uint32(c) - uint32('A')) >> 8) & 0xFF) ^ 0xFF) & (((((uint32('Z') - uint32(c)) >> 8) & 0xFF) ^ 0xFF) & (c - 'A'))) |
		(((((uint32(c) - uint32('a')) >> 8) & 0xFF) ^ 0xFF) & (((((uint32('z') - uint32(c)) >> 8) & 0xFF) ^ 0xFF) & (c - ('a' - 26)))) |
		(((((uint32(c) - uint32('0')) >> 8) & 0xFF) ^ 0xFF) & (((((uint32('9') - uint32(c)) >> 8) & 0xFF) ^ 0xFF) & (c - ('0' - 52))) |
		(((0 - (uint32(c) ^ uint32('+'))) >> 8) & 0xFF) ^ 0xFF) & 62) |
		(((0 - (uint32(c) ^ uint32('/'))) >> 8) & 0xFF) ^ 0xFF) & 63)
	return x | (((((0 - (uint32(x) ^ uint32(0))) >> 8) & 0xFF) ^ 0xFF) & (((((0 - (uint32(c) ^ uint32('A'))) >> 8) & 0xFF) ^ 0xFF) ^ 0xFF))
}


func main(){
    b64CharToByte(0)
}
